django/core/management/commands/dumpdata.py

```diff
@@ -184,7 +198,36 @@ def get_objects(count_only=False):
             if output and self.stdout.isatty() and options['verbosity'] > 0:
                 progress_output = self.stdout
                 object_count = sum(get_objects(count_only=True))
-            stream = open(output, 'w') if output else None
+            if output:
+                file_root, file_ext = os.path.splitext(output)
+                compression_formats = {
+                    '.gz': (gzip.open, {}, output),
+                    '.bz2': (open, {}, file_root),
+                    '.lzma': (open, {}, file_root),
+                    '.xz': (open, {}, file_root),
+                    '.zip': (open, {}, file_root),
```

Comment:
should this be `zipfile.Zipfile.open` or similar? Or removed (from here and my suggested docs)?

Reply:
Its lack in the documentation is deliberate. We have [already discussed](https://github.com/django/django/pull/13797#discussion_r547835369) with Nick about Zip archive support which totally differs from other compressed files. We left the previously agreed behavior to remove the extension and write the uncompressed data.